### ISSUE TITLE
Unexpected token � in JSON at position 0

### ISSUE THREAD

**Issue by mohd-akram** [microsoft/vscode-test#158]
To reproduce:

1. `yo code`
2. `cd ext`
3. `npm test`

```
Downloaded VS Code into /Users/user/Desktop/ext/.vscode-test/vscode-darwin-1.69.2
objc[23438]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x7ffb55368ec8) and /Users/user/Desktop/ext/.vscode-test/vscode-darwin-1.69.2/Visual Studio Code.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libGLESv2.dylib (0x1119079c8). One of the two will be used. Which one is undefined.
[main 2022-07-29T12:19:15.229Z] update#ctor - updates are disabled by the environment
2022-07-29 16:19:17.548 Code Helper (Renderer)[23441:980710] CoreText note: Client requested name ".NewYork-Regular", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:].
2022-07-29 16:19:17.549 Code Helper (Renderer)[23441:980710] CoreText note: Set a breakpoint on CTFontLogSystemFontNameRequest to debug.
[main 2022-07-29T12:19:18.496Z] Starting extension host with pid 23486 (fork() took 3 ms).
Loading development extension at /Users/user/Desktop/ext
Unexpected token � in JSON at position 0

  Extension Test Suite
    ✔ Sample test
  1 passing (220ms)
[main 2022-07-29T12:19:22.144Z] Waiting for extension host with pid 23486 to exit.
[main 2022-07-29T12:19:22.226Z] Extension host with pid 23486 exited with code: 0, signal: null.
Exit code:   0
Done
```

**Comment by seanwu1105** [comment]
I have the same problem with the same way to reproduce. Sometimes, it shows different error messages.

```
  Extension Test Suite
    ✔ Sample test
  1 passing (1ms)
[main 2022-08-07T03:44:10.137Z] Waiting for extension host with pid 68994 to exit.
[main 2022-08-07T03:44:10.154Z] Extension host with pid 68994 exited with code: 0, signal: null.
Unexpected token G in JSON at position 0
Exit code:   0
Done
```

```
  Extension Test Suite
    ✔ Sample test
  1 passing (1ms)
[main 2022-08-07T03:46:00.886Z] Waiting for extension host with pid 69872 to exit.
Cannot call write after a stream was destroyed: Error: Cannot call write after a stream was destroyed
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at errorBuffer (node:internal/streams/writable:517:14)
    at afterWrite (node:internal/streams/writable:501:5)
    at onwrite (node:internal/streams/writable:477:7)
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:91:12)
[main 2022-08-07T03:46:00.894Z] Extension host with pid 69872 exited with code: 0, signal: null.
Exit code:   0
Done
```

```
  Extension Test Suite
    ✔ Sample test
  1 passing (1ms)
[main 2022-08-07T03:45:42.531Z] Waiting for extension host with pid 69584 to exit.
Cannot call write after a stream was destroyed: Error: Cannot call write after a stream was destroyed
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at errorBuffer (node:internal/streams/writable:517:14)
    at afterWrite (node:internal/streams/writable:501:5)
    at onwrite (node:internal/streams/writable:477:7)
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:91:12)
[main 2022-08-07T03:45:42.541Z] Extension host with pid 69584 exited with code: 0, signal: null.
disposed: Error: disposed
    at S.requestPromise (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:24:50737)
    at Object.call (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:24:50574)
    at Proxy.<anonymous> (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:24:8764)
    at getEnvironment (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:81:7557)
    at Object.call (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:24:8381)
    at onPromise (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:23:4907)
    at onRawMessage (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:23:4283)
    at vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:23:3564
    at S.invoke (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:19:145)
    at m.deliver (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:19:2266)
    at w.fire (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:19:1844)
    at MessagePort.Z (vscode-file://vscode-app/Users/shuangwu/try-ext/.vscode-test/vscode-darwin-arm64-1.70.0/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:17:23629)
Exit code:   0
Done
```

Though the message does not fail the tests, it significantly pollute the log. I believe this is caused by some sort of race condition.

### Environment

- macOS 12.4
- engine.vscode: 1.69.0
- @vscode/test-electron: 2.1.5

**Comment by ssbarnea** [comment]
I seen the same in my own test suite and I was wondering where it was coming from. Hopefully someone would look at it and fix it. Noise running tests is dangerous.

**Comment by connor4312** [comment]
I've seen this before, but don't reproduce it in the latest 1.72.0. Please let me know if you hit it again.

**Comment by robertohuertasm** [comment]
HI @connor4312 I can still see this in 1.72.0.

**Comment by fearthecowboy** [comment]
I'm getting it on 1.76.1

**Comment by mohd-akram** [comment]
This seems to be fixed (tried with 1.85.2).
